src/19FAF0.c

```diff
@@ -223,7 +225,739 @@ s32 calc_player_test_enemy(void) {
+                            if ((battleStatus->currentAttackStatus & STATUS_FLAG_##STATUS_TYPE) && \
+                                try_inflict_status(target, STATUS_##STATUS_TYPE, STATUS_##STATUS_TYPE##_TURN_MOD)) { \
+                                tempBinary = TRUE; \
+                                wasStatusInflicted = TRUE; \
+                            } \
+
+                        INFLICT_STATUS(SHRINK);
+                        INFLICT_STATUS(POISON);
+                        INFLICT_STATUS(STONE);
+                        INFLICT_STATUS(SLEEP);
+                        INFLICT_STATUS(STOP);
+                        INFLICT_STATUS(STATIC);
+                        INFLICT_STATUS(FEAR);
+                        INFLICT_STATUS(PARALYZE);
+                        INFLICT_STATUS(DIZZY);
+
+                        #undef INFLICT_STATUS
+
+                        if (wasStatusInflicted) {
+                            if (dispatchEvent == EVENT_UNKNOWN_TRIGGER) {
+                                dispatchEvent = EVENT_HIT_COMBO;
+                            }
+
+                            if (dispatchEvent == EVENT_IMMUNE) {
+                                dispatchEvent = EVENT_HIT;
+                            }
+                        }
+                    }
+                }
+            } while (0);
+        }
+    }
+
+    battleStatus->wasStatusInflicted = wasStatusInflicted;
+    dispatch_event_actor(target, dispatchEvent);
+
+    if (!(target->flags & ACTOR_FLAG_NO_DMG_POPUP)) {
+        if (battleStatus->lastAttackDamage == 0) {
+            if (!tempBinary) {
+                if (!wasStatusInflicted) {
+                    func_8024EFE0(state->goalPos.x, state->goalPos.y, state->goalPos.z, 0, 1, 3);
+                }
+            }
+        } else if (!sp1C) {
+            if (battleStatus->currentAttackElement & (DAMAGE_TYPE_NO_OTHER_DAMAGE_POPUPS | DAMAGE_TYPE_SMASH)) {
+                show_damage_popup(state->goalPos.x, state->goalPos.y, state->goalPos.z, battleStatus->lastAttackDamage, 0);
+            } else {
+                func_802664DC(state->goalPos.x, state->goalPos.y, state->goalPos.z, battleStatus->lastAttackDamage, 0);
+            }
+
+            if (!(targetPart->targetFlags & 4)) {
+                func_802666E4(target, state->goalPos.x, state->goalPos.y, state->goalPos.z, battleStatus->lastAttackDamage);
+            }
+        }
+    }
+
+    if (tempBinary && gBattleStatus.flags1 & BS_FLAGS1_40 || gBattleStatus.flags1 & BS_FLAGS1_40) {
+        if (!(gBattleStatus.flags1 & BS_FLAGS1_80)) {
+            if (player->actorTypeData1[5]) {
+                sfx_play_sound_at_position(player->actorTypeData1[5], 0, state->goalPos.x, state->goalPos.y, state->goalPos.z);
+            }
+
+            if (battleStatus->lastAttackDamage > 0 && (sfx_play_sound(SOUND_231), battleStatus->lastAttackDamage > 0) ||    // TODO remove sfx_play_sound from conditional
```

Comment:
did you try doing 

```
if (damage > 0) {
    play_sound
    if (other thing)...
}
```

Reply:
Yeah I did buts it's doing two comparisons to damage > 0 in the assembly and I just didn't figure out how to get it to match

Reply:
ahh kk, makes sense